src/main/java/org/wso2/carbon/esb/connector/requests/SalesforceRequest.java

```diff
@@ -0,0 +1,362 @@
+    }
+
+    public void getQueryJobResultsAndStoreInFile(String queryJobId, String filePath, Integer maxRecords, String locator)
+            throws SalesforceConnectionException, InvalidConfigurationException {
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put(RequestConstants.HTTP_HEADER_AUTHORIZATION, RequestConstants.BEARER
+                + salesforceConfig.getAccessToken());
+        headers.put(RequestConstants.HTTP_HEADER_CONTENT_TYPE, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.HTTP_HEADER_ACCEPT, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.X_PRETTY_PRINT, "1");
+        RestRequest restRequest =
+                new RestRequest(HttpMethod.GET, SalesforceUtils.getGetQueryJobResultUrl(salesforceConfig, queryJobId,
+                        maxRecords, locator), null, headers);
+
+        restRequest.setOutputFilePath(filePath);
+        restRequest.setReceiveToFile(true);
+        sendRequest(restRequest);
+    }
+
+    public void getJobFailedResultsAndStoreInFile(String jobId, String filePath) throws SalesforceConnectionException, InvalidConfigurationException {
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put(RequestConstants.HTTP_HEADER_AUTHORIZATION, RequestConstants.BEARER
+                + salesforceConfig.getAccessToken());
+        headers.put(RequestConstants.HTTP_HEADER_CONTENT_TYPE, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.HTTP_HEADER_ACCEPT, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.X_PRETTY_PRINT, "1");
+        RestRequest restRequest =
+                new RestRequest(HttpMethod.GET, SalesforceUtils.getGetJobFailedResultsUrl(salesforceConfig, jobId),
+                        null, headers);
+
+        restRequest.setOutputFilePath(filePath);
+        restRequest.setReceiveToFile(true);
+        sendRequest(restRequest);
+    }
+
+    public String getJobFailedResults(String jobId) throws SalesforceConnectionException, InvalidConfigurationException {
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put(RequestConstants.HTTP_HEADER_AUTHORIZATION, RequestConstants.BEARER
+                + salesforceConfig.getAccessToken());
+        headers.put(RequestConstants.HTTP_HEADER_CONTENT_TYPE, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.HTTP_HEADER_ACCEPT, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.X_PRETTY_PRINT, "1");
+        RestRequest restRequest =
+                new RestRequest(HttpMethod.GET, SalesforceUtils.getGetJobFailedResultsUrl(salesforceConfig, jobId),
+                        null, headers);
+        return sendRequest(restRequest).getResponse();
+    }
+
+    public void getJobSuccessfulResultsAndStoreInFile(String jobId, String filePath) throws SalesforceConnectionException, InvalidConfigurationException {
```

Comment:
This seems to exceed the maximum line length. There we a few other places as well. Shall we format the code?

Reply:
Done